src/ngraph_builder.cc

```diff
@@ -3454,74 +3465,234 @@ static Status TranslateStridedSliceOp(
+  // For inclusive, the graph is clamped at 0 and dim-1
+  // Given dimension d, [0, d-1] are valid locations.
+  // -1 represents std::rend(). d represents std::end().
+  // These two are useful for representing exclusive boundaries for end-ranges
+  // Example for dim = 3:
+  // ranges:                 (-inf,-d)|   [-d,0)    |[0,d-1]|(d-1,inf)
+  // TF index:                  -5 -4 |-3  -2 -1    | 0 1 2 | 3 4 5
+  // clamped begin (inclusive):  0  0 | 0   1  2    | 0 1 2 | 3 3 3
+  // clamped end (exclusive):   -1 -1 | 0   1  2    | 0 1 2 | 3 3 3
+  auto clamper = [](int idx, size_t dim, bool inclusive) {
+    // if idx is in [-(d-1), d-1], then its same for both inclusive and
+    // exclusive
+    // The first 2 cases breaks down this range
+    if (idx >= 0 && idx <= (dim - 1)) {
+      return idx;
+    } else if (idx < 0 && idx + dim >= 0) {  // careful not to do idx >= -dim
+                                             // (since dim is unsigned)
+      return idx + (int)dim;  // Type casting to int to enable unambiguous auto
+                              // type inference of return type
+    } else if (idx > dim - 1) {
+      return (int)dim;
+    }  // The next case handles the clamping (differently for inclusive and
+       // exclusive cases)
+    else if (idx + dim <
+             0) {  // careful not to do idx < -dim (since dim is unsigned)
+      return 0 - (inclusive ? 0 : 1);
+    }
+  };
+
+  auto tf_to_ng = [clamper](int tf_begin_idx, int tf_end_idx, int tf_stride,
+                            size_t dim, bool begin_mask, bool end_mask,
+                            bool shrink_mask) {
+    // if begin mask is present, depending on stride sign use 0 (std::begin) or
+    // dim-1 (std::rbegin)
+    // clamped_end_idx could line in [-1, d]
+    int tf_ignore_begin_if_needed =
+        begin_mask ? (tf_stride > 0 ? 0 : dim - 1) : tf_begin_idx;
+    // if end mask is present, depending on stride sign use -1 (std::rend) or
+    // dim (std::end).
+    // However note, we cannot set to -1, since it has another meaning, hence
+    // setting to -(dim+1), which would translate to -1 in clamped coordinates
+    // take care to convert dim from sixze_t to int
+    int tf_ignore_end_if_needed =
+        end_mask ? (tf_stride > 0 ? dim : (-((int)dim + 1))) : tf_end_idx;
+
+    // using size_t for clamped_begin_idx because: clamped_begin_idx is
+    // inclusive, so it must lie in [0, dim-1]
+    size_t clamped_begin_idx = clamper(tf_ignore_begin_if_needed, dim, true);
+    int64 clamped_end_idx =
+        clamper(shrink_mask ? clamped_begin_idx + 1 : tf_ignore_end_if_needed,
+                dim, false);
+
+    // Now we have converted semantically non-monotonic and unbounded TF indexes
+    // (-inf, inf) to bounded and monotonic clamped indexes [-1, d]
+    // Now we need to convert clamped indexes [-1, d] to ngraph indexes [0, d]
+    // (taking care of reversal in case of negative strides)
+
+    bool needs_reverse = false;
```

Comment:
To be consistent with the tie conversion used later, can we change `needs_reverse` as an int type?